test/infrastructure/docker/cloudinit/runcmd.go

```diff
@@ -38,11 +40,7 @@ type Cmd struct {
 func (c *Cmd) UnmarshalJSON(data []byte) error {
 	// First, try to decode the input as a list
 	var s1 []string
-	if err := json.Unmarshal(data, &s1); err != nil {
-		if _, ok := err.(*json.UnmarshalTypeError); !ok {
```

Comment:
json-iterator does not provide a specific error when json can't be casted into the target type 😞 
same in unknown.go